code/modules/reagents/chemistry/reagents/pyrotechnic.dm

```diff
@@ -333,19 +333,26 @@
 	process_flags = ORGANIC | SYNTHETIC
 	taste_description = "bitterness"
 
+/datum/reagent/cryostylane/on_new(data)
+	..()
+	START_PROCESSING(SSprocessing, src)
+
+/datum/reagent/cryostylane/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
+
 /datum/reagent/cryostylane/on_mob_life(mob/living/M) //TODO: code freezing into an ice cube
 	if(M.reagents.has_reagent("oxygen"))
 		M.reagents.remove_reagent("oxygen", 1)
 		M.bodytemperature -= 30
 	return ..()
 
-/datum/reagent/cryostylane/on_tick()
-	if(holder.has_reagent("oxygen"))
-		holder.remove_reagent("oxygen", 2)
-		holder.remove_reagent("cryostylane", 2)
-		holder.temperature_reagents(holder.chem_temp - 200)
-		holder.temperature_reagents(holder.chem_temp - 200)
```

Comment:
These all did this twice because we didn't have the `processing` SS at the time, ergo processing only happened once every 2 seconds--so to compensate, I just called it twice. Having it be - 400 would not be the same as calling - 200 twice.